deploy/kubernetes/metricbeat-kubernetes.yaml

```diff
@@ -14,6 +14,13 @@ data:
       # Reload module configs as they change:
       reload.enabled: false
 
+    # To enable hints based autodiscover uncomment this:
+    #metricbeat.autodiscover:
+    #  providers:
+    #    - type: kubernetes
+    #      host: ${HOSTNAME}
```

Comment:
Is `$HOSTNAME` always set? and is it always set to a hostname in the list of nodes? (e.g. I think nodes can be registered with their IPs)

Alternatively we can add the `NODE_NAME` environment variable.
```
        - name: NODE_NAME
          valueFrom:
            fieldRef:
              fieldPath: spec.nodeName
```